plugin/impl.go

```diff
@@ -294,3 +320,57 @@ func getServerWithDefaults(server string, host string, protocol string) string {
 
 	return fmt.Sprintf("%s://%s", protocol, host)
 }
+
+func blockUntilBuildIsFinished(p *Plugin, client drone.Client, namespace, name string, buildNumber int) error {
+	fmt.Printf("\nblocking until triggered build is finished (canceling this build will cancel the downstream build too)\n")
+
+	timeout := time.After(p.settings.BlockTimeout)
+
+	//lint:ignore SA1015 refactor later
+	tick := time.Tick(10 * time.Second)
+
+	// listen for SIGINT and SIGTERM to cancel downstream build when our build is cancelled
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	defer close(sigs)
+
+	for {
+		select {
+		case <-sigs:
+			err := client.BuildCancel(namespace, name, int(buildNumber))
+			if err != nil {
+				return fmt.Errorf("could not cancel downstream job %d", buildNumber)
+			}
+
+			fmt.Printf("canceled downstream job %d\n", buildNumber)
+
+			return nil
+
+		// Got a timeout! fail with a timeout error
+		case <-timeout:
+			return fmt.Errorf("timed out waiting for %d", buildNumber)
+
+		// Got a tick, we should check on the build status
+		case <-tick:
+			build, err := client.Build(namespace, name, buildNumber)
+			if err != nil {
+				return err
+			}
+
+			s := build.Status
```

Comment:
could probably clean this up with a switch

```
switch build.Status {
case drone.StatusError, drone.StatusKilled, ...:
  return ...
case drone.statusPending:
  return nil
}
```
```

Reply:
Yeah, makes sense